File: 백준/Bronze/25304. 영수증/영수증.java
import java.util.*;

public class Main
{
	public static void main(String[] args) {
	    
	    long x;
	    int n, a, b;
	    
	    Scanner in = new Scanner (System.in);
	    
	    x = in.nextLong();
	    n = in.nextInt();
	    
	    int sum = 0;
	    for(int i=0; i<n; i++) {
	        
	        a = in.nextInt();
	        b = in.nextInt();
	        sum+=(a*b);
	    }
	   
	    if(x==sum) {
		    System.out.println("Yes");
	    }else{
	        System.out.println("No");
	    }
    }
}

File: 백준/Bronze/25304. 영수증/영수증.cc
#include <iostream>

using namespace std;

int main()
{
    unsigned int x, a;
    int n, b;
    int price, sum = 0;
    
    cin >> x >> n ;
    
    
    for (int i=1; i<n+1; i++) {
        cin >> a >> b;
        price = a*b;
        sum=sum+price;
    }
    
    if(x==sum) {
        cout << "Yes";
    }else {
        cout << "No";
    }
    
    return 0;
}